src/MainWindow.cpp

```diff
@@ -0,0 +1,936 @@
+    invalidKeyBox.setStandardButtons(QMessageBox::Ok);
+
+    int invalidKeyBoxSelection = invalidKeyBox.exec();
+
+    switch(invalidKeyBoxSelection)
+    {
+    // If the user presses the "Ok" button
+    case QMessageBox::Button::Ok :
+    {
+        // Terminate the application
+        emit conditionToTerminateMet();
+        break;
+    }
+
+    default :
+    {
+        // Terminate the application in the default case
+        emit conditionToTerminateMet();
+        break;
+    }
+    }
+}
+
+void MainWindow::createTableWidgetUsingMap()
+{
+    // Set the number of rows in the table widget to be the number of bills stored in the bill map
+    m_billTableWidget->setRowCount(m_billMap.size());
+
+    // Set the number of columns to the number of fields which each Bill displays
+    m_billTableWidget->setColumnCount(4);
+
+    // Set the table widget headers to the appropriate fields each Bill displays
+    m_billTableWidget->setHorizontalHeaderLabels(QString(m_BILL_NAME_COLUMN_HEADER_STRING + ";" + m_BILL_AMOUNT_DUE_COLUMN_HEADER_STRING + ";" + m_BILL_DUE_DATE_COLUMN_HEADER_STRING + ";" + m_BILL_FUNDING_STATUS_COLUMN_HEADER_STRING).split(";"));
+
+    // Initialize the row we're setting to zero
+    int row = 0;
+
+    // Create an iterator to iterate over the bill map
+    QMap<QString, Bill>::iterator billMapIterator;
+
+    for(billMapIterator = m_billMap.begin(); billMapIterator != m_billMap.end(); ++billMapIterator)
+    {
+        // Retrieve the current Bill being checked
+        Bill currentBill = *billMapIterator;
+
+        // Create a date edit for displaying and allowing the editing of the bill's due date
+        QDateEdit *dateEdit = new QDateEdit(this);
+
+        // Allow the display of a calendar for editing purposes
+        dateEdit->setCalendarPopup(true);
+
+        // Set the date edit based on the bill's due date
+        dateEdit->setDate(currentBill.getDueDate());
+
+        QComboBox *fundedStatusBox = new QComboBox(this);
+        fundedStatusBox->addItem(m_NOT_FUNDED_STRING);
+        fundedStatusBox->addItem(m_FUNDED_STRING);
+        currentBill.isFunded() ? fundedStatusBox->setCurrentIndex(1) : fundedStatusBox->setCurrentIndex(0);
+
+        // Set the columns appropriately to the Bill's attributes
+        m_billTableWidget->setItem(row, 0, new QTableWidgetItem(currentBill.getName()));
+        m_billTableWidget->setItem(row, 1, new QTableWidgetItem(QString::number(currentBill.getAmountDue(), 'f', 2)));
+        m_billTableWidget->setCellWidget(row, 2, dateEdit);
+        m_billTableWidget->setCellWidget(row, 3, fundedStatusBox);
+
+        // Increment the row for the next Bill
+        row++;
+    }
+}
+
+QDate MainWindow::convertDateStringToDate(QString p_dateString)
+{
+    // Convert the due date string into a date object and return it
+    return QDate::fromString(removeSpaces(p_dateString),"M/d/yyyy");
+}
+
+void MainWindow::readConfigAndCreateUI()
```

Comment:
This function has **way** too many decision points, break some of them out into helper method(s).

Reply:
Helper method added.